pkg/resource/deploy/step.go

```diff
@@ -337,26 +337,34 @@ func (s *DeleteStep) Logical() bool           { return !s.replacing }
 func (s *DeleteStep) Apply(preview bool) (resource.Status, StepCompleteFunc, error) {
 	// Refuse to delete protected resources (unless we're replacing them in
 	// which case we will of checked protect elsewhere)
-	if !s.replacing && s.old.Protect {
+	if !s.replacing && (s.old.Protect || s.old.DeleteBehaviour == resource.DeleteBehaviourProtect) {
 		return resource.StatusOK, nil, fmt.Errorf("unable to delete resource %q\n"+
 			"as it is currently marked for protection. To unprotect the resource, "+
```

Comment:
Does the message to "remove `protect` flag" still apply if the cause is DeleteBehaviorProtect?